src/util.c

```diff
@@ -17,3 +17,36 @@ long parse_int(char *str) {
 
     return parsed;
 }
+
+unsigned long parse_buttons_numbers(char *str,
+ignore_buttons_t *ignore_buttons) {
+    ignore_buttons->buttons = (unsigned int *)calloc(10, sizeof(unsigned int));
+    // no check for "more then UCHAR_MAX"
+    unsigned char max_symbols = 10;
+
+    char *strtok_value = strtok(str, ",");
+    while (strtok_value != NULL) {
+        // realloc
+        if (ignore_buttons->count >= max_symbols - 1) {
+            max_symbols += max_symbols;
+            unsigned int *p = (unsigned int *)realloc(ignore_buttons->buttons,
+                max_symbols * sizeof(unsigned int));
+            if (p == NULL) {
+                free(ignore_buttons->buttons);
+                bail("Cannot reallocate memory for ignore-buttons\n");
+                return ULONG_MAX;
+            } else {
+                ignore_buttons->buttons = p;
+            }
+        }
+
+        long number = atol(strtok_value);
+        ignore_buttons->buttons[ignore_buttons->count++] = number;
+        strtok_value = strtok(NULL, ",");
+    };
+
+    if (!ignore_buttons->count) {
+        return ULONG_MAX;
```

Comment:
I think this should never realistically happen, so I would bail here as well. Then we can also remove the `ULONG_MAX` returns and instead `assert(false)` since it will be a dead code path, right? The caller won't need to handle this since we'll always be bailing if something goes wrong.